### ISSUE TITLE
[Bug]:  Tsconfig not found issue during build

### ISSUE THREAD

**Issue by Snake-Master** [web-infra-dev/rsbuild#3994]
### Version

```sh
System:
    OS: Linux 5.14 CentOS Linux 7 (Core)
    CPU: (2) x64 Intel(R) Xeon(R) CPU E5-4650 v2 @ 2.40GHz
    Memory: 2.94 GB / 3.84 GB
    Container: Yes
    Shell: 4.2.46 - /bin/bash
  npmPackages:
    @rsbuild/core: ^1.0.19 => 1.1.2 
    @rsbuild/plugin-react: ^1.0.6 => 1.0.7
```

### Details

![Image](https://github.com/user-attachments/assets/79fc523b-9f4a-4376-8eb8-d414a83208d9)

### Reproduce link

无需仓库，官网的例子, 没有任何修改

### Reproduce Steps

直接使用 pnpm create rsbuild@latest 的 react typescript 的模板生成项目。

未作任何修改 pnpm install 后执行 pnpm build 就报 Tsconfig not found 编译不下去。

**Comment by chenjiahan** [comment]
Ref: https://github.com/web-infra-dev/rsbuild/discussions/3985

It looks like there is a problem with the Rspack resolver on Centos 7.

Can you try downgrading `@rspack/core` to 1.1.0 or 1.0.14? This will help us identify the version that introduced the regression.

For example, use pnpm overrides:

```json
// package.json
{
  "pnpm": {
    "overrides": {
      "@rspack/core": "1.0.14"
    }
  }
}

**Comment by Snake-Master** [comment]
> Ref: [#3985](https://github.com/web-infra-dev/rsbuild/discussions/3985)
> 
> It looks like there is a problem with the Rspack resolver on Centos 7.
> 
> Can you try downgrading `@rspack/core` to 1.1.0 or 1.0.14? This will help us identify the version that introduced the regression.
> 
> For example, use pnpm overrides:
> 
> // package.json
> {
>   "pnpm": {
>     "overrides": {
>       "@rspack/core": "1.0.14"
>     }
>   }
> }

The build was completed when the version of @rspack/core was downgraded to 1.1.0.  tks for your support.

**Comment by chenjiahan** [comment]
Other related issues:

- https://github.com/web-infra-dev/rspress/discussions/1593
- https://github.com/web-infra-dev/rsbuild/discussions/3985

**Comment by ayangweb** [comment]
![Image](https://github.com/user-attachments/assets/b018334e-c496-4c8f-8873-869a501ce330)

I am experiencing this with rspress, how do I downgrade?

**Comment by chenjiahan** [comment]
The same as https://github.com/web-infra-dev/rsbuild/issues/3994#issuecomment-2480882524

**Comment by ayangweb** [comment]
> The same as [#3994 (comment)](https://github.com/web-infra-dev/rsbuild/issues/3994#issuecomment-2480882524)

Okay, thanks. That'll do.

**Comment by chenjiahan** [comment]
I have released https://github.com/web-infra-dev/rsbuild/releases/tag/v1.1.3 to pin Rspack 1.1.0 and this should fix the problem.

**Comment by ayangweb** [comment]
That's fine. No problem.

**Comment by chenjiahan** [comment]
Fixed in Rspack v1.1.2
